packages/aws-cdk-lib/custom-resources/lib/provider-framework/waiter-state-machine.ts

```diff
@@ -172,7 +175,13 @@ export class WaiterStateMachine extends Construct {
       resources: ['*'],
     }));
 
-    const logGroup = logOptions?.destination ?? new LogGroup(this, 'LogGroup');
+    const logGroupName =
+      FeatureFlags.of(this).isEnabled(cxapi.WAITER_STATE_MACHINE_LOG_GROUP_NAME)
+        ? `/aws/vendedlogs/states/waiter-state-machine-${this.isCompleteHandler.functionName}-${this.node.addr}`
```

Comment:
`node.addr` is used to prevent duplicate names no matter how many times this construct is created in the stack.
Also, by using the auto-generated name of [the Lambda](https://github.com/aws/aws-cdk/blob/277023ccd5c9d34001daeeb20f6a6b061ae38bf3/packages/aws-cdk-lib/custom-resources/lib/provider-framework/provider.ts#L206) created in the `Provider` that calls this `WaiterStateMachine` construct, even if the `Provider` is deleted and then created again, the log group name will not duplicate previously created one. (This is because that lambda will be re-created again with a random name.)